pkgs/tools/graphics/gnuplot/default.nix

```diff
@@ -58,6 +62,9 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
+  # Set TEXINPUTS to include us
+  setupHook = lib.optionalString (withTeXLive) ./setup-hook.sh;
```

Comment:
If this setupHook is necessary for tests, then users won't be able to use `tikz` outside `nix-shell`, right? To fix this, replace the hook with makeWrapper.

Reply:
The problem is that gnuplot doesn't need TEXINPUTS set, but instead latex will need to know how to find gnuplot's tex files when building the output created using the tikz terminal.

That doesn't mean there isn't a cleaner way, which would be welcome! :)

Reply:
I have not tested this (except that it evaluates), but you can try:
```
-stdenv.mkDerivation rec {
+lib.fix (self: stdenv.mkDerivation rec {

-    ++ lib.optional withTeXLive (texlive.combine { inherit (texlive) scheme-small; })
+    ++ lib.optional withTeXLive (texlive.combine { inherit (texlive) scheme-small self; })

-    configureFlagsArray+=(--with-texdir=$out/share/texmf-nix/tex/latex/gnuplot)
+    configureFlagsArray+=(--with-texdir=$out/share/texmf/tex/latex/gnuplot)

-}
+})
```

Reply:
Hmm, if `runCommand "tikz-plot"` succeeds and it needs TEXINPUTS, then the wrapper should be an adequate replacement for setupHook, and it will also support users outside nix-shell. However, I'd prefer my snippet if it works.

Reply:
gnuplot does not invoke tex-- and so setting TEXINPUTS in the tex we pass it doesn't help.  Honestly I'm not entirely sure what purpose passing tex as a `buildInput` does, I'm not sure it's used at runtime.

The workflow is instead that gnuplot, when using the tikz terminal, generates output that is "incomplete" and needs gnuplot's generated files to be properly build using texlive.
So when the user includes this output in a document (`tikz-pdf` generates a basic example of this) the `tex` used needs to know how to find these pieces, hence the setup-hook.
If a wrapper were used, it would need to wrap tex or something, I think.

`tikz-plot` doesn't need any tex bits, or TEXINPUTS to be set, but rather it's that TEXINPUTS is how the `tex` in `tikz-pdf` knows to resolve the gnuplot tikz package/extensions.

At first I wanted to do something like:

```nix
texlive.combine { inherit (texlive) scheme-small; inherit gnuplot_full; }
```
For this purpose (the tex in `tikz-pdf`, not input to `gnuplot` build) which maybe would remove the need for the setup-hook and should work outside of nix-shell ... if it evaluated O:).

Unfortunately that's not really how `texlive.combiine` is meant to work, so instead of faking a texlive package for gnuplot or something I went the setup-hook route.

Make sense?